### DISCUSSION TITLE
passing acebase between files?

### DISCUSSION THREAD

**Discussion by pmioduszewski** [appy-one/acebase#90] [Q&A] [answered]
Hi,

I want to use acebase locally in two separate files, what's the best approach to achieve this?

index.js
```js
const { AceBase } = require("acebase");
const acebase = new AceBase("database");

const AcebaseAdapter = require("../utils/acebaseAdapter");
const useAcebaseSessions = (bot, session) =>
	bot.use(
		session({
			initial: () => new Object(),
			storage: new AcebaseAdapter(acebase),
		})
	);

module.exports = {
	useAcebaseSessions,
};
```
../utils/acebaseAdapter.js
```js
class AcebaseAdapter {
	constructor({ acebase, }) {
		this.acebase = acebase;
	}

	async read(key) {
		await this.acebase.ref(`sessions/${key}`).get((snapshot) => {
			if (snapshot.exists()) {
				const data = snapshot.val();
				return data;
			} else return undefined;
		});
	}
}

module.exports = AcebaseAdapter;
```

result of `useAcebaseSessions`:
`Cannot read properties of undefined (reading 'ref')` // acebase === `undefined`

do I really need to create another instance with `new AceBase`?
I feel like I'm making some silly mistake here :)

**Comment by appy-one** [comment, marked as answer]
You are passing the AceBase instance into your adapter constructor instead of an object with an acebase property. Try `new AcebaseAdapter({ acebase})` to pass it into the constructor the way you're expecting it, OR adjust your constructor function to `constructor(acebase)`

**Reply by pmioduszewski** [reply]
YES! xD
U're the best! 🙌
